src/detect-engine.c

```diff
@@ -977,12 +978,18 @@ static void DetectBufferTypeFreeFunc(void *data)
                     sigmatch_table[map->transforms.transforms[i].transform].name);
             continue;
         }
-        sigmatch_table[map->transforms.transforms[i].transform].Free(NULL, map->transforms.transforms[i].options);
+        sigmatch_table[map->transforms.transforms[i].transform].Free(
+                de_ctx, map->transforms.transforms[i].options);
```

Comment:
Looks ok but feels painful

Why will we need this `de_ctx` when freeing ?

Reply:
The Luaxform will need the `de_ctx` to unregister the thread ctx (see #12425 detect-transform-luaxform.c, function DetectTransformLuaxformFree)

Reply:
What is this thread ctx ?

Reply:
Another option would be to have `sigmatch_table ` having 2 different function pointers `Free` and 
`FreeCtx` (not sure it is better, but requires less modifications)